coopr-server/src/main/java/co/cask/coopr/store/entity/BaseSQLEntityStoreView.java

```diff
@@ -172,21 +173,24 @@ protected PreparedStatement getSelectStatement(Connection conn, EntityType entit
     return statement;
   }
 
-  private PreparedStatement getSelectAllStatement(Connection conn, EntityType entityType) throws SQLException {
-    String entityTypeId = entityType.getId();
-    // immune to sql injection since everything is an enum or constant
-    StringBuilder queryStr = new StringBuilder();
-    queryStr.append("SELECT ");
-    queryStr.append(entityTypeId);
-    queryStr.append(" FROM ");
-    queryStr.append(entityTypeId);
-    queryStr.append("s WHERE tenant_id=?");
+  private PreparedStatement getSelectAllLatestStatement(Connection conn, EntityType entityType) throws SQLException {
```

Comment:
this works right now, but is not efficient. It will also break if the "latest" version is not the max version. For example, if we want to support rolling back to an earlier version, we might want to tag the earlier version as "latest", but keep the higher version around.  Can we open a jira and reference it in a comment here to improve this logic.